meetings/2021-07/july-15.md

```diff
@@ -0,0 +1,734 @@
+
+CP: but I'm a journey probably not economically viable but let your it's yeah I mean you're gonna. 
+
+MM: Yeah. Yeah. Thanks I want to address exactly the issue you're raising. There is an existing pattern that one cannot you does not solve the problem, but I think the same as which I'll explain in a moment what that existing pattern But my question to you is the current the circumstances under which that pattern does not solve the problem. I believe your scenario equally does not solve the problem. Okay. The pattern that many projects use including the SES shim itself is that it is assumes that it's running first in its realm that the realm has not yet been corrupted at the time that it is running its module initialization. So during module initialization, it grabs the things that needs to remain. That it's going to then apply later and and the SES shim actually doesn't need to be as aggressive on that for security reasons, but it's aggressive about it anyway. In order for shim fidelity with spec, the spec specifies blah blah uses an internal method. It doesn't use the current binding, and the only way shim can emulate that is to grab the original binding before or anything gets corrupted, so it can continue to use the original binary. So that's a common practice, many systems of use that for years. The counter argument. Is that well, what if it doesn't run first and if that and I don't see another counter argument to it...
+
+JHD: that's that's not the problem I'm trying to solve, okay. So I think it is impossible to avoid having that problem. Once once you don't run first, all bets off. 
+
+MM: Good. 
+
+CP: Even the Realms can...
+
+JHD: Yeah you can you can modify because the the way into Realms which is pasta but you're also it's about my concern is about reducing the footprint scope of what I have to cash or hold onto that first run. So that it's all available later and the original Realms gives me one thing I have to grab this currently, I have a library that has 15 million downloads a week that I use for all my shims, that grabs everything and I have to explicitly, add each thing I need like, and then all of the shims have to like yeah it's just it's a it's a convenience and kind of scalability issue whereas Whereas, with the realm approach, I can it and of one and done to some degree. 
```

Comment:
```suggestion
JHD: My concern is about reducing the footprint and scope of what I have to cache or hold onto in that first-run code so that it's all available later, and the original Realms gives me one thing I have to grab. Currently, I have a library that has 15 million downloads a week that I use for all my shims, that grabs everything and I have to explicitly, add each thing I need, and then all of the shims have to load them all. It's a convenience and kind of scalability issue whereas, with the realm approach, I can do it "one and done" to some degree. 
```